packages/components/organisms/f-content-cards/src/components/ContentCards.js

```diff
@@ -215,251 +96,102 @@ export default {
+         * @param card
+         */
+        handleCardView (card) {
+            this.adapters.forEach(adapter => {
+                if (adapter.source === card.source) {
+                    adapter.handleCardView(card);
+                }
+            });
+        }
+    },
+
+    mounted () {
+        this.loggingKey = `f-content-cards--${this.userId}--${this.tags}`;
+        this.defaultLoggingData = {
+            data: {
+                key: this.loggingKey
+            }
+        };
+        this.runAdapterCallbacks((source, cards) => {
+            this.state = STATE_DEFAULT;
+            this.$emit(HAS_LOADED, { adapter: source, cards });
             this.$log.info(
-                `Content Cards (customContentCards) - Attempting to load cards via custom source. Key: (${this.loggingKey})`,
+                `Content Cards Adapters (${source}) - Successfully received content cards. Key: (${this.loggingKey})`,
                 this.logTags,
                 {
                     ...this.defaultLoggingData,
                     Count: cards.length,
                     data: {
                         ...this.defaultLoggingData.data,
-                        source: CARDSOURCE_CUSTOM
+                        source
                     }
                 }
             );
-            this.contentCards({
-                source: CARDSOURCE_CUSTOM
-            }, cards);
-        },
-
-        /**
-         * Takes appropriate response for click event for given card object based on its source
-         * @param card
-         */
-        handleCardClick (card) {
-            switch (card.source) {
-                case CARDSOURCE_METADATA:
-                    this.trackMetadataCardClick(card);
-                    this.brazeAdapter.logCardClick(card.id);
-                    break;
-                case CARDSOURCE_CUSTOM:
-                    this.trackCustomCardClick(card);
-                    break;
-                default:
-                    this.$log.error(
-                        `Content Cards (handleCardClick) - Invalid card source type. Key: (${this.loggingKey})`,
-                        null,
-                        this.logTags,
-                        {
-                            ...this.defaultLoggingData,
-                            data: {
-                                ...this.defaultLoggingData.data,
-                                source: CARDSOURCE_METADATA
-                            }
+        }, (source, error) => {
+            if (error.type === 'NoCards') {
+                this.$emit(HAS_LOADED, { adapter: source, cards: [] });
+                this.$log.info(
+                    `Content Cards Adapters (${source}) - No content cards. Key: (${this.loggingKey})`,
+                    this.logTags,
+                    {
+                        ...this.defaultLoggingData,
+                        data: {
+                            errorMessage: error
```

Comment:
Might be good to add `source` as a data value into the no cards / error states as well?